### DISCUSSION TITLE
flex library

### DISCUSSION THREAD

**Discussion by pierrehub2b** [airsdk/Adobe-Runtime-Support#3207] [General]
Hello, I need your help 

I am compiling a component library using ant and compc
Here is the xml I use to compile my lib : 

```
	<target name="compile.project" description="" depends="clean.release.folder,init.sdk">
		<compc compiler.as3="true" 
			compiler.debug="false" 
			compiler.optimize="true" 
			compiler.accessible="false" 
			output="release/ui-components.swc" 
			incremental="false"
			fork="true" 
			maxmemory="512m">

			<source-path path-element="src" />
			<source-path path-element="embed" />
			
			<include-sources dir="src" includes="*" />
			
		</compc>
	</target>
```

the result is a swc file of 2 800 Mo size

If I build this project with FlashBuilder I have a swc file with 980 Mo size.

If I compare the 2 swc's there is a 'mx' and a 'spark' folder in the swc file compiled with ant, this 2 folders are not present in the swc file built by FlashBuilder.

How can I have the same swc file using ant and compc ? wich oprions should I use with compc ?

Pierre

**Comment by pierrehub2b** [comment]
Not 2 800 Mo and 980 Mo size sorry ... but 2 800 Kb and 890 Kb

**Comment by ajwfrost** [comment]
I suspect the fact you've got mx/spark classes included in your library file is because of that `source-path` element.. depending on what's in the `embed` folder? Have you got the Flex source code included either in your main `src` folder, on in `embed`?

It's also then worth checking the configuration file that you're loading (probably flex-config.xml) to see if that has got any "include library" requests.. it shouldn't do, for the Flex mx/spark classes, these should only be used as references for compilation.

Adobe's Flex documentation has gone into a single big PDF: https://help.adobe.com/archive/en_US/flex/using/flex_4.6_help.pdf
but there should be information about the command-line options: e.g. instead of `source-path` it should be something like `external-library-path` pointing at the Flex libraries (SDKfolder/frameworks/libs)?

thanks

**Reply by hydroper** [reply]
Josh has mentioned me Flex's HTML version in [archive.org](https://web.archive.org/web/20160321225840/http://help.adobe.com/en_US/flex/using/index.html)
